File: app/src/main/java/crabster/rudakov/sberschoollesson30hwk/domain/repositories/IDictionaryRepository.java
package crabster.rudakov.sberschoollesson30hwk.domain.repositories;

import androidx.annotation.*;

import java.util.*;

import crabster.rudakov.sberschoollesson30hwk.domain.repositories.models.*;
import io.reactivex.*;

public interface IDictionaryRepository {

    Completable add(@NonNull DictionaryItemModel dictionaryItemModel);

    Single<List<DictionaryItemModel>> getList();

    Completable delete(long id);

}

File: app/src/main/java/crabster/rudakov/sberschoollesson30hwk/domain/repositories/IDictionaryRepository.kt
package crabster.rudakov.sberschoollesson30hwk.domain.repositories

import crabster.rudakov.sberschoollesson30hwk.domain.repositories.models.DictionaryItemModel
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Single

interface IDictionaryRepository {

    fun add(dictionaryItemModel: DictionaryItemModel): Completable

    fun getList(): Single<List<DictionaryItemModel>>

    fun delete(id: Long): Completable

}